core/src/Streamly/Internal/Data/Binary/Encode.hs

```diff
@@ -243,6 +251,147 @@ word64leD w = D.Stream step W64B1
 word64le :: Monad m => Word64 -> Stream m Word8
 word64le = fromStreamD . word64leD
 
+{-# INLINE int8 #-}
+int8 :: Int8 -> Stream m Word8
+int8 i = Stream.fromPure (fromIntegral i :: Word8)
```

Comment:
This could be `word8 (fromIntegral i :: Word8)` and similarly for other int types as well.